docs: Add `DestroyRef` and `takeUntilDestroyed` sections to lifecycle doc (#50042)
angular/angular: aio/content/guide/lifecycle-hooks.md

The `ngOnDestroy()` method is also the time to notify another part of the application that the component is going away.

### DestroyRef

> As an alternative to `ngOnDestroy()`, you can also inject Angular's `DestroyRef` and register callback functions to be called when the enclosing context is destroyed. This can be useful for building reusable utilities that require cleanup.
> 

Reviewer: ```suggestion
In addition to to `ngOnDestroy()`, you can inject Angular's `DestroyRef` and register callback functions to be called when the enclosing context is destroyed. This can be useful for building reusable utilities that require cleanup.

```
I would use "in addition to" rather than "As an alternative" since you can use both at the same time.

Register a callback with the `DestroyRef`:

```ts
@Component(...)

> class CounterCmp {

Reviewer: ```suggestion
class Counter {
```

  count = 0;

>   constructor(destroyRef: DestroyRef) {
>     // Start a timer to increment the counter every second.
>     const id = setInterval(() => this.count++, 1000);
> 
>     // Stop the timer when the component is destroyed.
>     destroyRef.onDestroy(() => clearInterval(id));
>   }

Reviewer: ```suggestion
  constructor() {
    // Start a timer to increment the counter every second.
    const id = setInterval(() => this.count++, 1000);

    // Stop the timer when the component is destroyed.
    const destroyRef = inject(DestroyRef);
    destroyRef.onDestroy(() => clearInterval(id));
  }
```
`inject`?

}
```

Like `ngOnDestroy`, `DestroyRef` works in any Angular service, directive, component, or pipe.

### `takeUntilDestroyed`

<div class="alert is-important">

> `takeUntilDestroyed` is available for [developer preview](https://angular.io/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.

Reviewer 2: Nit: the link could be relative : 
```suggestion
`takeUntilDestroyed` is available for [developer preview](/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.
```

Author: Thanks!

</div>

> When using RxJS Observables in components, a common pattern is to apply an operator which will complete the Observable when the component is destroyed.
> 
> Angular's `@angular/core/rxjs-interop` package provides a convenient `takeUntilDestroyed` operator to simply this common task:

Reviewer: ```suggestion
When using RxJS Observables in components or directives, you may want to complete any observables when the component or directive is destroyed. Angular's `@angular/core/rxjs-interop` package provides an operator, `takeUntilDestroyed`, to simply this common task:
```

```ts
data$ = http.get('...').pipe(takeUntilDestroyed());
```

> By default `takeUntilDestroyed` must be run in an injection context, and will use the current `DestroyRef`. If an injection context isn't available, `DestroyRef` can also be provided explicitly.

Reviewer: ```suggestion
By default, `takeUntilDestroyed` must be called in an injection context so that it can access `DestroyRef`. If an injection context isn't available, you can explicitly provide a `DestroyRef`.
```
